pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -268,7 +275,7 @@ private void receiveMessageFromConsumer(ConsumerImpl<T> consumer) {
             } else {
                 // Call receiveAsync() if the incoming queue is not full. Because this block is run with
                 // thenAcceptAsync, there is no chance for recursion that would lead to stack overflow.
-                receiveMessageFromConsumer(consumer);
+                receiveMessageFromConsumer(consumer, messages.size() > 0);
```

Comment:
If entered receiveMessageFromConsumer from this line and the messages.size() = 0 in this cycle, is there any chance to call batch receive in the future?

Reply:
Yes, if `messages.size() = 0`, the current round of receive messages from the internal consumer will use `consumer.receiveAsync()`. After the internal consumer has new incoming messages, the next round will use the `batchReceiveAsync()` again.